kmip/tests/unit/core/attributes/test_digest.py

```diff
@@ -316,9 +318,17 @@ def test_not_equal_on_not_equal(self):
             digest_value=self.digest_value_b,
             key_format_type=self.key_format_type_b)
         b = Digest()
+        c = Digest(
+            hashing_algorithm=self.hashing_algorithm_d,
+            digest_value=self.digest_value_b,
+            key_format_type=self.key_format_type_c)
+        d = Digest(
+            key_format_type=self.key_format_type_c)
 
         self.assertTrue(a != b)
         self.assertTrue(b != a)
+        self.assertTrue(b != c)
+        self.assertTrue(b != d)
```

Comment:
These tests were meant to specifically run through the code paths on the __eq__ function of the Digest class ( line ~444 in the core/attributes file). This comparator has an elif chain for comparing objects. This specific combination of comparison objects ensured that all comparison paths in the code were tested. The values were selected according to what was needed for the comparisons.